pygmt/datatypes/header.py

```diff
@@ -0,0 +1,278 @@
+        newdims = {dim: dim for dim in dims}
+        # Loop over dimensions and get the dimension name and attributes from header
+        for dim, nameunit in zip(dims, nameunits, strict=False):
+            # The long_name and units attributes.
+            long_name, units = _parse_nameunits(nameunit.decode())
+            if long_name:
+                attrs[dim]["long_name"] = long_name
+            if units:
+                attrs[dim]["units"] = units
+
+            # "degrees_east"/"degrees_north" are the units for geographic coordinates
+            # following CF-conventions
+            if units == "degrees_east":
+                attrs[dim]["standard_name"] = "longitude"
+                newdims[dim] = "lon"
+            elif units == "degrees_north":
+                attrs[dim]["standard_name"] = "latitude"
+                newdims[dim] = "lat"
+
+            # Axis attributes are "X"/"Y"/"Z"/"T" for horizontal/vertical/time axis.
+            attrs[dim]["axis"] = dim.upper()
+            idx = 2 if dim == "y" else 0
+            attrs[dim]["actual_range"] = np.array(self.wesn[idx : idx + 2])
+
+        # Save the lists of dimension names and attributes in the _nc attribute.
+        self._nc = {
+            "dims": [newdims[dim] for dim in dims],
+            "attrs": [attrs[dim] for dim in dims],
+        }
+
+    def get_name(self) -> str:
+        """
+        Get the name of the grid from the grid header.
+
+        Returns
+        -------
+        name
+            The name of the grid.
+        """
+        return "z"
+
+    def get_data_attrs(self) -> dict:
+        """
+        Get the attributes for the data variable from the grid header.
+
+        Returns
+        -------
+        attrs
+            The attributes for the data variable.
+        """
+        attrs: dict[str, Any] = {}
+        attrs["Conventions"] = "CF-1.7"
+        attrs["title"] = self.title
+        attrs["history"] = self.command
+        attrs["description"] = self.remark
+        long_name, units = _parse_nameunits(self.z_units.decode())
+        if long_name:
+            attrs["long_name"] = long_name
+        if units:
+            attrs["units"] = units
+        attrs["actual_range"] = np.array([self.z_min, self.z_max])
+        return attrs
+
+    def get_dims(self):
+        """
+        Get the dimension names from the grid header.
+
+        Returns
+        -------
+        dims : tuple
+            The dimension names.
+        """
+        if not hasattr(self, "_nc"):
+            self._parse_dimensions()
+        return self._nc["dims"]
+
+    def get_dim_attrs(self) -> list:
+        """
+        Get the attributes for each dimension from the grid header.
+
+        Returns
+        -------
+        attrs
+            List of attributes for each dimension.
+        """
+        if not hasattr(self, "_nc"):
+            self._parse_dimensions()
+        return self._nc["attrs"]
+
+    def get_gtype(self) -> int:
```

Comment:
How about using the `@property` decorator, similar to https://github.com/GenericMappingTools/pygmt/blob/bbf909d9ea6e1cf48af3566e490b14d10de09785/pygmt/figure.py#L121-L129

This would become:

```suggestion
    @property
    def gtype(self) -> int:
```

Reply:
I was thinking about using `@property` too. But when we use `header.gtype`, it's not straightforward to know if `gtype` is a field or a property of the `GMT_GRID_HEADER` class, although technically it makes no differences.

Reply:
Yeah, either way, we're just accessing the grid header's attributes. I'd prefer adding the `@property` decorator to make it clear that we are accessing a property/attribute, but up to you.

Reply:
Change them to properties in 79f4c879e1cc0af2ce3f8cbd9d5eaac6a5644ad6.